r/src/RecordBatch.cpp

```diff
@@ -96,23 +97,31 @@ int RecordBatch__to_file(const std::shared_ptr<arrow::RecordBatch>& batch,
 }
 
 // [[Rcpp::export]]
-RawVector RecordBatch__to_raw(const std::shared_ptr<arrow::RecordBatch>& batch) {
-  MemoryPool* pool = default_memory_pool();
+RawVector RecordBatch__to_stream(const std::shared_ptr<arrow::RecordBatch>& batch) {
+  std::unique_ptr<io::MockOutputStream> mockSink;
+  mockSink.reset(new io::MockOutputStream());
 
-  std::shared_ptr<ResizableBuffer> buffer;
-  R_ERROR_NOT_OK(AllocateResizableBuffer(pool, 0, &buffer));
+  std::shared_ptr<arrow::ipc::RecordBatchWriter> mockWriter;
+  R_ERROR_NOT_OK(arrow::ipc::RecordBatchStreamWriter::Open(mockSink.get(),
+                                                           batch->schema(),
+                                                           &mockWriter));
```

Comment:
Use the function `arrow::ipc::WriteRecordBatchStream({batch}, &mock_writer)` to save yourself about 3 lines.